vars/parseStageInfo.groovy

```diff
@@ -267,6 +267,26 @@ def call(Map config = [:]) {
       }
     }
 
+    String provider
+    // Highest priority is TestProvider parameter
+    if (startedByUser() && params.TestProvider && params.TestProvider != "") {
+      provider = params.TestProvider
+    } else {
+      // Next highest priority is a stage specific Test-provider-*
+      provider = cachedCommitPragma("Test-provider" + result['pragma_suffix'], null)
+      if (!repeat) {
+        // Followed by the more general Test-provider:
+        provider = cachedCommitPragma("Test-provider", null)
```

Comment:
```suggestion
      provider = cachedCommitPragma("Test-provider" + result['pragma_suffix'], cachedCommitPragma("Test-provider", null))
```

Reply:
Applied to both 'Test-provider' and 'Test-repeat'